### PR TITLE
Network: broadcast TransactionPayloadQuery to all private TX participants

### LINKED ISSUE #720: Network: broadcast TransactionPayloadQuery to all participants in a private TX
Currently the last non-self PAL participant is selected, but this is vague, ambiguous and not spec'd. Best option would probably be to just broadcast the request to all (except the local node) participants in the PAL header.

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes https://github.com/nuts-foundation/nuts-node/issues/737
Fixes #720 

Based on https://github.com/nuts-foundation/nuts-node/pull/712

- [x] Integration test reports "already connected to peer XYZ", check whether that issues already existed previously.

### CODE REVIEW

File: network/transport/v2/protocol.go [hunk 1]
```diff
@@ -198,9 +198,17 @@ func (p *protocol) handlePrivateTxRetryErr(hash hash.SHA256Hash) error {
...
+		return err
+	}
+	if nodeDID.Empty() {
+		return errors.New("node DID is not set")
+	}
+
 	epal := dag.EncryptedPAL(tx.PAL())
 
-	pal, senderDID, err := p.decryptPAL(epal)
+	pal, err := p.decryptPAL(epal, nodeDID)
```
Comment (woutslakhorst) [inline network/transport/v2/protocol.go:211]: reason to duplicate nodeDID check (also in decryptPAL)

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: network/transport/v2/handlers.go [hunk 2]
```diff
@@ -54,7 +54,15 @@ func (p *protocol) handleTransactionPayloadQuery(peer transport.Peer, msg *Trans
 		}
 		epal := dag.EncryptedPAL(tx.PAL())
 
-		pal, _, err := p.decryptPAL(epal)
+		nodeDID, err := p.nodeDIDResolver.Resolve()
```
[the remark this answers is not in the document]
Comment (reinkrul) [reply network/transport/v2/handlers.go]: Reverted it.

**Comment by reinkrul** [comment]
@woutslakhorst ready for rereview.

File: network/transport/v2/protocol.go [hunk 3]
```diff
@@ -214,19 +212,27 @@ func (p *protocol) handlePrivateTxRetryErr(hash hash.SHA256Hash) error {
...
-	err = conn.Send(p, &Envelope{Message: &Envelope_TransactionPayloadQuery{
-		TransactionPayloadQuery: &TransactionPayloadQuery{
-			TransactionRef: tx.Ref().Slice(),
-		},
-	}})
-	if err != nil {
-		return fmt.Errorf("failed to send TransactionPayloadQuery message(tx=%s, DID=%s): %w", hash.String(), senderDID, err)
+	if !sent {
+		return fmt.Errorf("unable to retrieve private TX payload, no connections to any of the participants (tx=%s, PAL=%v)", hash.String(), pal)
 	}
```
Comment (dmeijboom) [inline network/transport/v2/protocol.go:237]: Could be removed I think as there is already a log entry for when send fails

**Review by dmeijboom — APPROVED** [review]

File: network/transport/v2/protocol.go (hunk 3 above)
Comment (woutslakhorst) [reply network/transport/v2/protocol.go:237]: The error return is logged as error, so the error formatting here is ok. The first part of the sentence is a bit of a duplicate. The wrapper adds `retry of TransactionPayloadQuery failed: %v`

**Review by woutslakhorst — APPROVED** [review]
With a single side note on the specific log message

**Pull request merged**